### ISSUE TITLE
💅 `useHeadingContent` false positive on Astro headings using `set:html` / `set:text`

### ISSUE THREAD

**Issue by abemedia** [biomejs/biome#11644]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.12
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_DISTRIBUTION:                npm
  BIOME_LOG_FILE:                    unset
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  BIOME_BINARY:                      unset
  RUST_BACKTRACE:                    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.20.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.33.0

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

</details>

### Rule name

useHeadingContent

### Playground link

https://biomejs.dev/playground/?tab=formatter&pane=Diagnostics&language=astro#code=LQAtAC0ACgBjAG8AbgBzAHQAIABoAGUAYQBkAGkAbgBnACAAPQAgACcASABlAGwAbABvACAAdwBvAHIAbABkACcAOwAKAC0ALQAtAAoAPABoADEAIABzAGUAdAA6AGgAdABtAGwAPQB7AGgAZQBhAGQAaQBuAGcAfQAgAC8APgAKADwAaAAyACAAcwBlAHQAOgB0AGUAeAB0AD0AewBoAGUAYQBkAGkAbgBnAH0AIAAvAD4ACgA8AGgAMwAgAHMAZQB0ADoAaAB0AG0AbAA9AHsAaABlAGEAZABpAG4AZwB9AD4APAAvAGgAMwA%2BAA%3D%3D

### Expected result

`set:html` and `set:text` on a heading should count as content, the same way
`dangerouslySetInnerHTML` does for JSX, so none of the three headings above should be
reported. Vue's `v-html` / `v-text` are probably the same situation for `.vue` files, though
I have only verified Astro.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by abemedia** [comment]
Wow, thanks for the speedy resolution @ematipico!
